components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/dao/ClaimDAO.java

```diff
@@ -232,4 +233,80 @@ protected void deleteClaimProperties(Connection connection, int claimId, int ten
             throw new ClaimMetadataException("Error while deleting claim properties", e);
         }
     }
+
+    /**
+     * Deletes a set of claim properties.
+     *
+     * @param connection         Database connection
+     * @param claimId            ID of the claim
+     * @param claimPropertyNames Names of the claim properties to be deleted
+     * @param tenantId           Tenant ID
+     * @throws ClaimMetadataException if deletion fails
+     */
+    protected void deleteClaimProperties(Connection connection, int claimId,
+                                         Set<String> claimPropertyNames, int tenantId) throws ClaimMetadataException {
+
+        String query = SQLConstants.DELETE_CLAIM_PROPERTY_BY_NAME;
+        try (PreparedStatement prepStmt = connection.prepareStatement(query)) {
+            prepStmt.setInt(1, claimId);
+            prepStmt.setInt(3, tenantId);
+            
+            for (String propertyName : claimPropertyNames) {
+                prepStmt.setString(2, propertyName);
+                prepStmt.addBatch();
+            }
+            prepStmt.executeBatch();
+        } catch (SQLException e) {
+            throw new ClaimMetadataException("Error while deleting claim properties: " + claimPropertyNames, e);
+        }
+    }
+
+    /**
+     * Updates claim properties atomically by performing deletions and additions within a single transaction.
+     * This method ensures that all operations either complete successfully or are rolled back entirely.
+     *
+     * @param connection              Database connection to be used
+     * @param claimId                 ID of the claim whose properties are being updated
+     * @param newClaimProperties      Map of new claim properties to be added (property name to value mapping)
+     * @param claimPropertiesToDelete Set of claim property names to be deleted
+     * @param tenantId                ID of the tenant
+     * @throws ClaimMetadataException If an error occurs during the transaction or while managing claim properties
+     */
+    protected void updateClaimPropertiesAtomically(Connection connection, int claimId,
+                                                   Map<String, String> newClaimProperties,
+                                                   Set<String> claimPropertiesToDelete,
+                                                   int tenantId) throws ClaimMetadataException {
```

Comment:
Is this formatting intentional?

Reply:
Resolved via https://github.com/wso2/carbon-identity-framework/pull/6113/commits/68fc46c2ac1fdba49735c501e43f2dc0371ae302